keras/src/saving/saving_lib.py

```diff
@@ -444,12 +463,8 @@ def _load_model_from_fileobj(fileobj, custom_objects, compile, safe_mode):
                             pathlib.Path(extract_dir.name, _VARS_FNAME_H5),
                             mode="r",
                         )
-                    except OSError:
-                        # Fall back when it is a read-only system
-                        weights_store = H5IOStore(_VARS_FNAME_H5, zf, mode="r")
-                else:
-                    # Fall back when `fileobj` is an `io.BytesIO`. Typically,
-                    # this usage is for pickling.
+                except Exception:
+                    # Revert to the previous behavior.
```

Comment:
"If we can't use the local disk for any reason, read the weights from the zip archive on the fly, which is less efficient"

Reply:
Fixed.